### PR TITLE
Exraction implementation with tika, Fast api endpoint integration, docker and compose infra addedImpleem

### PR DESCRIPTION
**Pull request by gmuditdce**
Implementation for #2  #3  #4 #5 #11 
- Extraction Rest api (fast api) implmenation
- Tika integration for extraction of content and metadata
-  docker and compose infra
- Readme updated

### CODE REVIEW

**Review by rajmbcoderx — APPROVED** [review]
Looks good, I will help you next week to add unit tests and integration tests

File: cdr_content_extraction/api/routes/ExtractionResponse.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from enum import Enum
+import json
+
+from pydantic import BaseModel, Field
+
```
Comment (ggrig) [inline cdr_content_extraction/api/routes/ExtractionResponse.py:5]: Although these are just stubs for several classes, we cannot put all those into a single file
Please create a separate file for each of the classes. 
The file name must be the same as the class name
For instance MetaDataResponse.py should contain the MetaDataResponse class definition

File: cdr_content_extraction/extraction-config.yml [hunk 2]
```diff
@@ -0,0 +1,261 @@
+---
+service: text_extraction
+tools:
+  apache_tika:
+    content_type:
+      - application/java-vm
```
Comment (ggrig) [inline cdr_content_extraction/extraction-config.yml:6]: Why do we need all those content types?

File: cdr_content_extraction/extraction/ExtractionConfig.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    INITIALIZED = False
+
+    def __init__(self):
+        pass
+
+    def init(self):
+        ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
+        with open(os.path.join(ROOT_DIR, '../extraction-config.yml'), "r") as stream:
+            try:
+                config = yaml.safe_load(stream)
```
Comment (ggrig) [inline cdr_content_extraction/extraction/ExtractionConfig.py:20]: We use environment variables for configuration parameters
Please refer to how it is structured in https://github.com/filetrust/cdr-content-integrity/blob/main/cdr_content_management/common_settings/Config.py
Please utilize that file for handling config paramaters

**Review by ggrig — CHANGES_REQUESTED** [review]

**Comment by ggrig** [comment]
The PR contains stubs for some classes.
This may be a good start point.
However, I am not sure if you should call those as an implementation of any of the tasks
It would make more sense if you could concentrate on a complete implementation of a single task per PR

**Comment by chrisdiao** [comment]
@gmuditdce , could you please add version label.

**Comment by rajmbcoderx** [comment]
@dinis-cruz-gw  (@gmuditdce @ggrig Merging this PR
We will have something that is working and extracting content to test.
We will see how we can incorporate some review comments @ggrig has on the PR in later PR, this one is getting big and should be a good starting point to build upon

**Pull request merged**